pdr_backend/ppss/abstract/base_ss.py

```diff
@@ -0,0 +1,158 @@
+from abc import ABC
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from enforce_typing import enforce_types
+
+from pdr_backend.cli.arg_feed import ArgFeed, ArgFeeds
+from pdr_backend.cli.arg_pair import ArgPair
+from pdr_backend.subgraph.subgraph_feed import SubgraphFeed
+
+
+class MultiFeedSS(ABC):
+    FEEDS_KEY = ""
+
+    @enforce_types
+    def __init__(self, d: dict, assert_feed_attributes: Optional[List] = None):
+        assert self.__class__.FEEDS_KEY
+        self.d = d
+        # save self.exchs_dict
+        self.exchs_dict: dict = {}  # e.g. {"binance" : ccxt.binance()}
```

Comment:
There is no need for most ss classes to know anything about ccxt exchanges. The ccxt info only be seen by ohlcv data factory and maybe its ss.